src/app/components/shared/date-time-filter/filter.component.ts

```diff
@@ -0,0 +1,199 @@
+   * @param formEndDate A date object of type `NgbDate` supplied by the bootstrap date input representing the end date
+   * @returns Whether the start date comes before the end date, or the two dates are equal
+   */
+  public dateIsInBoundary(formStartDate: NgbDate, formEndDate: NgbDate): boolean {
+    // if the user has not set a start date and end date, we know that the date range is not defined and therefore, cannot be out of bounds
+    if (!isInstantiated(formStartDate) || !isInstantiated(formEndDate)) {
+      return true;
+    }
+
+    // converting to a date object allows the comparison of dates using boolean operators
+    const startDate = new Date(formStartDate.year, formStartDate.month, formStartDate.day);
+    const endDate = new Date(formEndDate.year, formEndDate.month, formEndDate.day);
+
+    return startDate <= endDate;
+  }
+
+  private generateFilters(model: Model): Filters | null {
+    const filter: InnerFilter = {};
+
+    if (this.site) {
+      filter["sites.id"] = { eq: this.site.id };
+    } else if (this.region) {
+      filter["regions.id"] = { eq: this.region.id };
+    } else if (this.project) {
+      filter["projects.id"] = { eq: this.project.id };
+    }
+
+    this.setDateFilter(filter, model);
+    this.setTimeOfDayFilter(filter, model);
+
+    if (JSON.stringify(filter) === JSON.stringify({})) {
+      return null;
+    }
+
+    if (JSON.stringify({ filter }) !== this.previousFilters) {
+      this.previousFilters = JSON.stringify({filter });
+      return { filter };
+    }
+
+    return null;
+  }
+
+  private setDateFilter(
+    filter: InnerFilter,
+    model: Model
+  ): void {
+    const filterStartDate = model?.dateStartedAfter;
+    const filterEndDate = model?.dateFinishedBefore;
+
+    // if there is no dates specified or the date boundaries are not correct, bail out
+    if (
+      !model?.dateFiltering ||
+      (!filterStartDate && !filterEndDate) ||
+      !this.dateIsInBoundary(filterStartDate, filterEndDate)
+    ) {
+      return;
+    }
+
+    if (!this.isInvalidDate(model.dateStartedAfter) && model.dateStartedAfter) {
+      filter["recordedDate"] ??= {};
+      filter["recordedDate"].greaterThanOrEqual =
+        `${this.formatter.format(model.dateStartedAfter)}T00:00:00.000Z`;
+    }
+
+    if (!this.isInvalidDate(model.dateFinishedBefore) && model.dateFinishedBefore) {
+      filter["recordedEndDate"] ??= {};
+      (filter["recordedEndDate"] as Comparisons).lessThanOrEqual =
+        // to allow users to input the same date as the starting and ending date, it is required that time is included
+        // if only the date is used (not with T24:00:00), it will default to T00:00:00, causing 1 day filters not to work
+        // if a time of day filter is specified, it will take priority over the time set in the date, causing no breakage in time filtering
+        `${this.formatter.format(model.dateFinishedBefore)}T23:59:59.000Z`;
+    }
+  }
+
```

Comment:
hint: the code written here was done before support for arrays was adding to the filtering API: https://github.com/QutEcoacoustics/baw-server/wiki/API:-Filtering

![image](https://user-images.githubusercontent.com/1452968/220526440-ad11a8b1-d261-4fef-b5ee-ed3240dc7503.png)